tensorflow/c/c_api.h

```diff
@@ -520,16 +520,23 @@ typedef struct TF_AttrMetadata {
 // Returns metadata about the value of the attribute `attr_name` of `oper`.
 TF_CAPI_EXPORT extern TF_AttrMetadata TF_OperationGetAttrMetadata(
     TF_Operation* oper, const char* attr_name, TF_Status* status);
+TF_CAPI_EXPORT extern TF_AttrMetadata TF_OperationGetAttributeMetadata(
```

Comment:
This needs documentation of why we have duplicated all these functions in the API.

It's also not super clear to me why do we need to duplicate here.

Reply:
My understanding here is that the C API is going to have the same issues as the C++ API, namely that the C API will not disambiguate between `double`s and `float`s correctly (`2.0` is considered to be a `double`)

Reply:
Right but this particular method doesn't have float or double in the signature.

Reply:
Right, this is for API consistency, since we would be deprecating the `GetAttr` version of the API?

Reply:
And while it doesn't have it in the signature, the behaviour of the function changes (albeit only slightly) when we disambiguate `double` and `float`.